app/src/main/java/ardjomand/leonardo/arcmovies/model/UpcomingMovie.java

```diff
@@ -0,0 +1,248 @@
+package ardjomand.leonardo.arcmovies.model;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by unity on 25/02/18.
+ */
+
+public class UpcomingMovie implements Parcelable {
```

Comment:
Is this object really completely different from movie details? Specially from a domain perspective?

Reply:
For this specific application, the main difference is in the way the genres are sent by each API:
MovieDetails stores the genres as List<Genre> genres
UpcomingMovie stores the genres as List<Integer> genreIds

Perhaps the repository could deserialize the data into these two POJOs, depending on which API is being used, but then convert them to a single Movie POJO made specifically for the View and Presenter.